src/client/cockpit-beiboot

```diff
@@ -0,0 +1,312 @@
+        command('cockpit.report-exists', {name: os.path.exists(name) for name in files})
+    """,
+    **ferny.BEIBOOT_GADGETS
+}
+
+
+class DefaultRoutingRule(RoutingRule):
+    peer: Peer
+
+    def __init__(self, router: Router, peer: Peer):
+        super().__init__(router)
+        self.peer = peer
+
+    def apply_rule(self, options: dict[str, object]) -> Peer:
+        return self.peer
+
+    def shutdown(self) -> None:
+        self.peer.close()
+
+
+class AuthorizeResponder(ferny.InteractionResponder):
+    router: Router
+
+    def __init__(self, router: Router, payload: bytes):
+        self.router = router
+        self.payload = payload
+
+    async def do_askpass(self, messages: str, prompt: str, hint: str) -> Optional[str]:
+        if hint == 'none':
+            # We have three problems here:
+            #
+            #   - we have no way to present a message on the login
+            #     screen without presenting a prompt and a button
+            #   - the login screen will not try to repost the login
+            #     request because it doesn't understand that we are not
+            #     waiting on input, which means that it won't notice
+            #     that we've logged in successfully
+            #   - cockpit-ws has an issue where if we retry the request
+            #     again after login succeeded then it won't forward the
+            #     init message to the client, stalling the login.  This
+            #     is a race and can't be fixed without -ws changes.
+            #
+            # Let's avoid all of that by just showing nothing.
+            return None
+
+        challenge = 'X-Conversation - ' + base64.b64encode(prompt.encode()).decode()
+        response = await self.router.request_authorization(challenge,
+                                                           messages=messages,
+                                                           prompt=prompt,
+                                                           hint=hint,
+                                                           echo=False)
+
+        b64 = response.removeprefix('X-Conversation -').strip()
+        passwd = base64.b64decode(b64.encode()).decode()
+        logger.debug('Returning a %d chars password', len(passwd))
+        return passwd
+
+    async def do_custom_command(self, command: str, args: tuple, fds: list[int], stderr: str) -> None:
+        logger.debug('Got ferny command %s %s %s', command, args, stderr)
+
+        if command == 'beiboot.provide':
+            size, = args
+            assert size == len(self.payload)
+            assert isinstance(self.router, SshBridge)
+            assert self.router.ssh_peer.transport is not None
+            self.router.ssh_peer.transport.write(self.payload)
+
+        elif command == 'cockpit.report-exists':
+            file_status, = args
+            # Constructed to throw KeyError if an unexpected file is requested
+            # ...please don't look at this code...
```

Comment:
Can you prefix with `HACK:` so that we retain a greppable list of things to clean up later™?

Reply:
gern.